### PR TITLE
[DROOLS-6859] Update bootable jar 

### PR DESCRIPTION
**Pull request by Rikkola**
[DROOLS-6859](https://issues.redhat.com/browse/DROOLS-6859)

* https://github.com/kiegroup/kie-wb-distributions/pull/1156
* https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1908

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -38,7 +38,7 @@
     <errai.marshalling.very_short_names>false</errai.marshalling.very_short_names>
     <!-- The version greater than 1.0.0.GA is not compatible with GWT 2.8.x -->
     <version.javax.validation>1.0.0.GA</version.javax.validation>
-    <wildfly.bootable.jar.gav>org.wildfly:wildfly-galleon-pack:${version.org.wildfly}</wildfly.bootable.jar.gav>
+    <wildfly.bootable.jar.gav>org.jboss.eap:wildfly-galleon-pack:4.0.0.GA-redhat-00002</wildfly.bootable.jar.gav>
```
Comment (mareknovotny) [inline pom.xml:41]: this should be defined/specified in productized profile i think

Comment (Rikkola) [reply pom.xml:41]: True. I will look into it.

Comment (rgdoliveira) [reply pom.xml:41]: I have a comment regarding the prod XP version that you are using. Prod version 4.0.0.GA-redhat-00002 is EAP XP4 which is not released yet (not available in MRRC), EAP QE is still testing it.

The latest prod version of XP public available is 3.0.0.GA-redhat-00012, which is on top of EAP 7.4.3, so I believe this is the version we should use currently.

Comment (rgdoliveira) [reply pom.xml:41]: I have a comment regarding the prod XP version that you are using. Prod version 4.0.0.GA-redhat-00002 is EAP XP4 which is not released yet (not available in MRRC), EAP QE is still testing it.

The latest prod version of XP public available is 3.0.0.GA-redhat-00012, which is on top of EAP 7.4.3, so I believe this is the version we should use currently.

Comment (Rikkola) [reply pom.xml:41]: @rgdoliveira Right, I have access for the XP4 and the goal is to update to XP4. This means this PR might be blocked until it is released, but would be helpful to run tests against the 4.0.0.GA-redhat-00002 if that is possible?

Comment (rgdoliveira) [reply pom.xml:41]: @Rikkola I agree, I guess this PR needs to be on hold until XP4 is released to MRRC.
Regarding running the tests against prod build 4.0.0.GA-redhat-00002, do you mean CI? (cc @Ginxo)

Comment (Rikkola) [reply pom.xml:41]: @rgdoliveira Yes manual testing passed, but it is more or less limited. So would be good to pick up any issues before the final is released and we might get it fixed for us.

Comment (Ginxo) [reply pom.xml:41]: @Rikkola do we keep this PR as draft? Thanks

**Review by mareknovotny — APPROVED** [review]

**Comment by rgdoliveira** [comment]
@Rikkola @mareknovotny  FYI, XP4 will used prod version **4.0.0.GA-redhat-00003** for CR2.

**Comment by rgdoliveira** [comment]
@Rikkola @mareknovotny  FYI, XP4 will used prod version **4.0.0.GA-redhat-00003** for CR2.

**Comment by mareknovotny** [comment]
jenkins do product fdb

**Comment by mareknovotny** [comment]
jenkins retest this please

**Pull request merged**